extensions/tls-registry/runtime/src/main/java/io/quarkus/tls/runtime/CertificateRecorder.java

```diff
@@ -188,4 +209,29 @@ public TlsConfigurationRegistry get() {
     public void register(String name, Supplier<TlsConfiguration> supplier) {
         register(name, supplier.get());
     }
+
+    static void addNamedConfigForForProvidersIfNecessary(Class<?> type, Map<String, TlsBucketConfig> namedConfigs) {
+        var container = Arc.container();
+        var instances = container.listAll(type);
```

Comment:
@mkouba is this the proper way? I always used `select`.

Reply:
`ArcContainer#listAll()` is just fine. And in this particular case it's an equivalent to `Instance.select().handles()`.